glean-core/rlb/src/lib.rs

```diff
@@ -137,9 +179,10 @@ pub fn initialize(cfg: Configuration, client_info: ClientInfoMetrics) {
 
         let glean = match Glean::new(core_cfg) {
             Ok(glean) => glean,
-            // glean-core already takes care of logging errors: other bindings
-            // simply do early returns, as we're doing.
-            Err(_) => return,
+            Err(err) => {
+                log::error!("Failed to initialize Glean: {}", err);
+                return
+            },
         };
 
         // glean-core already takes care of logging errors: other bindings
```

Comment:
Funny thing: until we get to this line, anything that calls `with_glean` may crash.

Reply:
That's true !